spikes/lk_core_touch_sensor/main.cpp

```diff
@@ -0,0 +1,146 @@
+	auto belt_right_back  = io::expanded::DigitalOut<> {io_expander, mcp23017::pin::PA1};
+	auto belt_right_front = io::expanded::DigitalOut<> {io_expander, mcp23017::pin::PA0};
+}	// namespace power_mode_pin
+
+namespace dac {
+	auto left  = CoreQDACMCP4728 {corei2c, 0xC2};
+	auto right = CoreQDACMCP4728 {corei2c, 0xC0};
+	namespace channel {
+		auto ear_left		  = mcp4728::channel::C;
+		auto ear_right		  = mcp4728::channel::C;
+		auto belt_left_back	  = mcp4728::channel::A;
+		auto belt_left_front  = mcp4728::channel::B;
+		auto belt_right_back  = mcp4728::channel::B;
+		auto belt_right_front = mcp4728::channel::A;
+	}	// namespace channel
+}	// namespace dac
+
+namespace sensor {
+	auto ear_left = CoreTouchSensor {detect_pin::ear_left, power_mode_pin::ear_left, dac::left, dac::channel::ear_left};
+	auto ear_right =
+		CoreTouchSensor {detect_pin::ear_right, power_mode_pin::ear_right, dac::right, dac::channel::ear_right};
+	auto belt_left_back	  = CoreTouchSensor {detect_pin::belt_left_back, power_mode_pin::belt_left_back, dac::left,
+											 dac::channel::belt_left_back};
+	auto belt_left_front  = CoreTouchSensor {detect_pin::belt_left_front, power_mode_pin::belt_left_front, dac::left,
+											 dac::channel::belt_left_front};
+	auto belt_right_back  = CoreTouchSensor {detect_pin::belt_right_back, power_mode_pin::belt_right_back, dac::right,
+											 dac::channel::belt_right_back};
+	auto belt_right_front = CoreTouchSensor {detect_pin::belt_right_front, power_mode_pin::belt_right_front, dac::right,
+											 dac::channel::belt_right_front};
+}	// namespace sensor
+}	// namespace touch
+
+auto sensor = CoreTouchSensor {touch::sensor::ear_left};
+
+auto resetByTouchAndRelease()
+{
+	log_info("Put your hand on the sensor then release !\n\n");
+	rtos::ThisThread::sleep_for(100ms);
+	while (sensor.read()) {
+		rtos::ThisThread::sleep_for(100ms);
+	}
+	log_info("OK !\n\n");
+}
+
+void calibration()
+{
+	log_info("Start calibration in 5s...");
+	log_info("Put your hand on the sensor\n\n");
+	rtos::ThisThread::sleep_for(5s);
+	auto value = uint16_t {CoreTouchSensor::default_min_sensitivity_value};
+	auto step  = uint16_t {10};
+
+	constexpr auto accurate_read_count = uint8_t {10};
+	auto read_count					   = uint8_t {0};
+
+	sensor.setSensitivity(value);
+
+	while (read_count < accurate_read_count && value <= CoreTouchSensor::default_max_sensitivity_value) {
+		if (!sensor.read()) {
+			read_count = 0;
+			value += step;
+			sensor.setSensitivity(value);
+			log_info(" Value : %d\n\n", value);
+
+		} else {
+			read_count++;
+		}
+		rtos::ThisThread::sleep_for(100ms);
+	}
+	sensor.setSensitivity(value);
+	log_info("Final sensitivity Value : %d\n\n", value);
+}
+
+auto main() -> int
+{
+	logger::init();
+
+	HelloWorld hello;
+	hello.start();
+	log_info("Hello, World!\n\n");
+
+	rtos::ThisThread::sleep_for(2s);
+
+	sensor.init();
+	sensor.setSensitivity(0x0FF0);
```

Comment:
this number doesn't mean much for someone reviewing the code. Is it a magic number? Is it "au doigt mouillé"? Is it high sensitivity or low sensitivity? why `0x0FF0` and not `0x0FF1`?

solutions:

1. create an `enum class Sensitivity` with different levels: `low/medium/high` so that it's easier to set/understand
1. use a named constant for the value --> `auto kDefault{High/Low}Sensitivity = 0x0FF0;`
1. have `setSensitivity(value)` take a float --> `setSensitivity(float value)` with values between 0.0 (super low sensitivity) to 1.0 (super high sensitivity) and do the `float --> uint16_t` mapping in the implementation

I favor `2.` and `3.` and they can be combined

This can be implemented in a followup PR.